content/02/02.04/script.js.adoc

```diff
@@ -1,3 +1,223 @@
+
+
+1 scenario (1 undefined)
+3 steps (3 undefined)
+0m00.000s
+npm ERR! Test failed.  See above for more details.
+----
+
+shot::[7]
+`Undefined` means Cucumber doesn’t know what to do for any of the three steps we wrote in our Gherkin scenario. It needs us to provide some step definitions.
+
+shot::[8]
+Step definitions translate from the plain language you use in Gherkin into Ruby code.
+
+shot::[9]
+When Cucumber runs a step, it looks for a step definition that matches the text in the step. If it finds one, then it executes the code in the step definition.
+
+If it doesn’t find one… well, you’ve just seen what happens. Cucumber helpfully prints out some code snippets that we can use as a basis for new step definitions.
+shot::[10]
+Let’s copy those.
+
+shot::[11]
+We’ll paste them into a JavaScript file under a new directory called `step_definitions` underneath the `features` directory. I’ll just call it `steps.rb`.
+
+shot::[12]
+Before running our process again, we need to import the `Given`,
+shot::[13]
+`When`
+shot::[14]
+and `Then` keywords from the `cucumber` package.
+
+.steps.js
+[source,js]
+----
+const { Given, When, Then } = require('cucumber');
+
+Given('Lucy is located 15m from Sean', function () {
+  // Write code here that turns the phrase above into concrete actions
+  return 'pending';
+});
+
+When('Sean shouts “free bagels at Sean’s”', function () {
+  // Write code here that turns the phrase above into concrete actions
+  return 'pending';
+});
+
+Then('Lucy hears Sean’s message', function () {
+  // Write code here that turns the phrase above into concrete actions
+  return 'pending';
+});
+----
+
+shot::[15]
+Now we run Cucumber again.
+
+shot::[16]
+This time the output is a little different. We now have a pending step and two skipped ones. This means Cucumber found all our step definitions, and executed the first one.
+shot::[17]
+But that first step definition returns `'pending'`, which causes Cucumber to stop, skip the rest of the steps, and mark the scenario as pending. 
+
+[source,bash]
+----
+$ npm test
+
+> shouty-js@1.0.0 test /home/fedex/cucumber/shouty-js
+> cucumber-js
+
+P--
+
+Warnings:
+
+1) Scenario: Listener is within range # features/hear_shout.feature:2
+   ? Given Lucy is located 15m from Sean # features/step_definitions/steps.js:5
+       Pending
+   - When Sean shouts “free bagels at Sean’s” # features/step_definitions/steps.js:10
+   - Then Lucy hears Sean’s message # features/step_definitions/steps.js:15
+
+1 scenario (1 pending)
+3 steps (1 pending, 2 skipped)
+0m00.000s
+npm ERR! Test failed.  See above for more details.
+
+----
+
+It’s time to start work on our solution. At this point we need to do a little domain modelling. The words in the step provide us some good hints about some names that will appear in our domain model. Our scenario mentions two people, Lucy and Sean. So it looks like we need a couple of person objects at least. Each person will also need to know their own location.
```

Comment:
I cut this stuff out of the Ruby script - it doesn't make sense to say it here anymore as we're not actually going to do any domain modelling until the next video.

Can you have a look at the Ruby video and make sure the flow matches?

Reply:
I think this bit says "Now that we’ve wired up our step definitions to the Gherkin steps, it’s almost time to start working on our solution. First though, let’s tidy up the generated code."